hazelcast/src/test/java/com/hazelcast/instance/DefaultAddressPickerInterfacesTest.java

```diff
@@ -0,0 +1,429 @@
+        assertThat(inetAddress.getHostAddress(), anyOf(equalTo("192.168.1.1"), equalTo("172.172.172.172")));
+    }
+
+    /**
+     * When: Network interface has both IPv4 and IPv6 addresses and IPv4 is preferred.<br/>
+     * Then: The IPv4 address is picked.
+     */
+    @Test
+    public void testIPv4Preferred() throws Exception {
+        List<NetworkInterface> networkInterfaces = new ArrayList<NetworkInterface>();
+        networkInterfaces.add(createNetworkConfig(
+                builder().withName("eth0").withAddresses("fe80::9711:82f4:383a:e254", "192.168.1.1", "::cace").build()));
+        when(NetworkInterface.getNetworkInterfaces()).thenReturn(Collections.enumeration(networkInterfaces));
+
+        InetAddress inetAddress = getInetAddressFromDefaultAddressPicker();
+        assertNotNull("Not-null InetAddress is expected", inetAddress);
+        assertEquals("192.168.1.1", inetAddress.getHostAddress());
+    }
+
+    /**
+     * When: Network interface has both IPv4 and IPv6 addresses and IPv6 is preferred.<br/>
+     * Then: The IPv6 address is picked.
+     */
+    @Test
+    public void testIPv6Preferred() throws Exception {
+        String origPref = setSystemProperty(SYS_PROP_PREFER_IPV4, "false");
+        try {
+            List<NetworkInterface> networkInterfaces = new ArrayList<NetworkInterface>();
+            networkInterfaces.add(createNetworkConfig(builder().withName("eth0")
+                    .withAddresses("fe80::9711:82f4:383a:e254", "172.17.0.1").build()));
+            when(NetworkInterface.getNetworkInterfaces()).thenReturn(Collections.enumeration(networkInterfaces));
+
+            InetAddress inetAddress = getInetAddressFromDefaultAddressPicker();
+            assertNotNull("Not-null InetAddress is expected", inetAddress);
+            assertEquals("fe80:0:0:0:9711:82f4:383a:e254", inetAddress.getHostAddress());
+        } finally {
+            setSystemProperty(SYS_PROP_PREFER_IPV4, origPref);
+        }
+    }
+
+    /**
+     * When: Multiple interfaces with different configuration is used, but only one IPv4 interface/address combination is
+     * pickable.<br/>
+     * Then: The correct address is picked.
+     */
+    @Test
+    public void testComplexScenario() throws Exception {
+        List<NetworkInterface> networkInterfaces = new ArrayList<NetworkInterface>();
+        networkInterfaces.add(
+                createNetworkConfig(builder().withName("lo").withLoopback(true).withAddresses("127.0.0.1", "::1").build()));
+        networkInterfaces
+                .add(createNetworkConfig(builder().withName("docker0").withUp(false).withAddresses("172.17.0.1").build()));
+        networkInterfaces.add(createNetworkConfig(builder().withName("wlp3s0").withUp(false).build()));
+        networkInterfaces.add(createNetworkConfig(
+                builder().withName("enp0s25").withAddresses("fe80::9711:82f4:383a:e254", "192.168.1.4").build()));
+        networkInterfaces
```

Comment:
I'm going to guess you wanted to make this interface virtual. Otherwise, I don't see any virtual interfaces being created.

Reply:
Good catch. I'm going to add 2 test methods with virtual Network Interfaces.